src/energyplus/Test/OutputTableSummaryReports_GTest.cpp

```diff
@@ -0,0 +1,118 @@
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include <gtest/gtest.h>
+#include "EnergyPlusFixture.hpp"
+
+#include "../ErrorFile.hpp"
+#include "../ForwardTranslator.hpp"
+#include "../ReverseTranslator.hpp"
+
+#include "../../model/Model.hpp"
+#include "../../model/OutputTableSummaryReports.hpp"
+
+#include "../../utilities/idf/IdfObject.hpp"
+#include "../../utilities/idf/IdfExtensibleGroup.hpp"
+
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/Output_Table_SummaryReports_FieldEnums.hxx>
+
+#include <vector>
+
+using namespace openstudio::energyplus;
+using namespace openstudio::model;
+using namespace openstudio;
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_OutputTableSummaryReports_NoOutputTableSummaryReports) {
+  Model model;
+
+  ForwardTranslator ft;
+  Workspace workspace = ft.translateModel(model);
+
+  std::vector<WorkspaceObject> idf_outputTableSummaryReports = workspace.getObjectsByType(IddObjectType::Output_Table_SummaryReports);
+  ASSERT_EQ(1u, idf_outputTableSummaryReports.size());
+  WorkspaceObject idf_outputTableSummaryReport(idf_outputTableSummaryReports[0]);
+
+  ASSERT_EQ(1u, idf_outputTableSummaryReport.numExtensibleGroups());
+  IdfExtensibleGroup eg = idf_outputTableSummaryReport.extensibleGroups()[0];
+  std::string s = eg.getString(Output_Table_SummaryReportsExtensibleFields::ReportName).get();
+  ASSERT_EQ(s, "AllSummary");
+}
+
+TEST_F(EnergyPlusFixture, ForwardTranslator_OutputTableSummaryReports_DefaultOutputTableSummaryReports) {
+  Model model;
+
+  OutputTableSummaryReports outputTableSummaryReports = model.getUniqueModelObject<OutputTableSummaryReports>();
+
+  ForwardTranslator ft;
+  Workspace workspace = ft.translateModel(model);
+
+  std::vector<WorkspaceObject> idf_outputTableSummaryReports = workspace.getObjectsByType(IddObjectType::Output_Table_SummaryReports);
+  ASSERT_EQ(1u, idf_outputTableSummaryReports.size());
+  WorkspaceObject idf_outputTableSummaryReport(idf_outputTableSummaryReports[0]);
+
+  ASSERT_EQ(1u, idf_outputTableSummaryReport.numExtensibleGroups());
+  IdfExtensibleGroup eg = idf_outputTableSummaryReport.extensibleGroups()[0];
+  std::string s = eg.getString(Output_Table_SummaryReportsExtensibleFields::ReportName).get();
+  ASSERT_EQ(s, "AllSummary");
```

Comment:
This test might need to change based on the comments I've made. User should have the possibility of completely disabling reports IMHO.